crates/ty_ide/src/semantic_tokens.rs

```diff
@@ -0,0 +1,2608 @@
+            ast::Expr::Lambda(lambda) => {
+                // Handle lambda parameters
+                if let Some(parameters) = &lambda.parameters {
+                    self.visit_parameters(parameters, None);
+                }
+
+                // Visit the lambda body
+                self.visit_expr(&lambda.body);
+            }
+            _ => {
+                // For all other expression types, let the default visitor handle them
+                walk_expr(self, expr);
+            }
+        }
+    }
+
+    /// Visit decorators, handling simple name decorators vs complex expressions
+    fn visit_decorator(&mut self, decorator: &ast::Decorator) {
+        match &decorator.expression {
+            ast::Expr::Name(name) => {
+                // Simple decorator like @staticmethod - use Decorator token type
+                self.add_token(
+                    name.range(),
+                    SemanticTokenType::Decorator,
+                    SemanticTokenModifier::empty(),
+                );
+            }
+            _ => {
+                // Complex decorator like @app.route("/path") - use normal expression rules
+                self.visit_expr(&decorator.expression);
+            }
+        }
+    }
+
+    fn visit_type_param(&mut self, type_param: &TypeParam) {
+        // Emit token for the type parameter name
+        let name_range = type_param.name().range();
+        self.add_token(
+            name_range,
+            SemanticTokenType::TypeParameter,
+            SemanticTokenModifier::DEFINITION,
+        );
+
+        // Visit bound expression (for TypeVar)
+        match type_param {
+            TypeParam::TypeVar(type_var) => {
+                if let Some(bound) = &type_var.bound {
+                    self.visit_type_annotation(bound);
+                }
+                if let Some(default) = &type_var.default {
+                    self.visit_type_annotation(default);
+                }
+            }
+            TypeParam::ParamSpec(param_spec) => {
+                if let Some(default) = &param_spec.default {
+                    self.visit_type_annotation(default);
+                }
+            }
+            TypeParam::TypeVarTuple(type_var_tuple) => {
+                if let Some(default) = &type_var_tuple.default {
+                    self.visit_type_annotation(default);
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::tests::cursor_test;
+
+    /// Helper function to assert exact token counts
+    ///
+    /// This function helps reduce boilerplate in tests by allowing you to specify
+    /// the expected counts for multiple token types in a single call.
+    ///
+    /// # Example
+    /// ```
+    /// assert_token_counts(&tokens, &[
+    ///     (SemanticTokenType::BuiltinConstant, 3),
+    ///     (SemanticTokenType::Variable, 3),
+    /// ]);
+    /// ```
+    fn assert_token_counts(
+        tokens: &SemanticTokens,
```

Comment:
This will ensure that Rust shows the callsite of `assert_token_counts` if the assertion fails instead of this function in a stacktrace, which is probably more useful (because it helps narrow down which test failed)

```suggestion
    #[track_caller]
    fn assert_token_counts(
        tokens: &SemanticTokens,
```